### PR TITLE
Issue/2977 add media settings back

### PR DESCRIPTION
**Pull request by tonyr59h**
This PR introduces a new category, `This Device`, which groups device-specific settings. `Enable Location` is the only Calypso setting that is device specific. Two settings that were in the old settings screen, `Default Image Width` and `Upload and link to full image`, are also added under the new category.

cc @kwonye (last issue!), @mattmiklic (design verification)

### CODE REVIEW

File: WordPress/src/main/res/layout/related_posts_dialog.xml [hunk 1]
```diff
@@ -42,22 +42,22 @@
             app:fontFamily="openSans"
             app:fontVariation="normal" />
 
-        <CheckBox
+        <org.wordpress.android.widgets.WPCheckBox
```
Comment (kwonye) [inline WordPress/src/main/res/layout/related_posts_dialog.xml]: Niicceee cc: @nbradbury :)

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -503,6 +507,8 @@
     <string name="site_settings_location_title">Enable Location</string>
     <string name="site_settings_default_category_title">Default Category</string>
     <string name="site_settings_default_format_title">Default Format</string>
+    <string name="site_settings_default_image_width_title">Default Image Width</string>
```
Comment (kwonye) [inline WordPress/src/main/res/values/strings.xml]: This is (almost exactly) the same text as `max_thumbnail_px_width`. Please consolidate.

**Comment by kwonye** [comment]
Noticed that `blog_preferences.xml` still contains `CheckBox`s, can we change them to `WPCheckBox` please?

Once all comments are addressed and design is :ok_hand: with @mattmiklic, feel free to :shipit:

Great work!

**Comment by mattmiklic** [comment]
Everything looks just fine visually. :ok_hand: 

I ran into a bug; I noticed that the first time the settings screen loads for a particular blog, I get a "The username or password you entered is incorrect" error. It doesn't appear on subsequent loads of the settings page for that site. If I switch to a different site, I'll get the same error the first time it loads. This is all on a fresh install of this branch with my WP.com credentials (which are obviously correct, since it was able to load my sites and their settings).

screenshot_20160106-235850

**Comment by kwonye** [comment]
By goodness @mattmiklic, you have a system update!!!

**Comment by tonyr59h** [comment]
> I ran into a bug

Resolved in 3aefb835e93add31eefc7d9fc3befdb5cd51f69a

**Comment by kwonye** [comment]
Verified repro steps from @tonyr59h for that bug

New account with no account cached. Resolved in 3aefb83

**Comment by kwonye** [comment]
Addresses #2977

**Pull request merged**